src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -86,9 +91,9 @@ internal class FoldingRangeEndpoint : IVSFoldingRangeEndpoint
 
         private async Task<IEnumerable<FoldingRange>?> HandleCoreAsync(RazorFoldingRangeRequestParam requestParams, DocumentContext documentContext, CancellationToken cancellationToken)
         {
+            var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Why pre-calculate code document / parse the file prior to delegation?

Reply:
I was noticing that FoldingRange requests were timing out and it APPEARED to be GetCodeDocument's fault because it was throwing TaskCancelled, but actually it was inside SendRequest which was not throwing when it timed out. This doesn't actually solve anything and is orthogonal to this PR, will remove.